core/src/main/java/org/apache/iceberg/lock/ServerSideHttpLockManager.java

```diff
@@ -0,0 +1,149 @@
+  private static final int REQUEST_SUCCESS = 200;
+  public static final String REQUEST_URL = "lock.http.conf.request.url";
+  private HttpClient httpClient = null;
+
+  public ServerSideHttpLockManager() {}
+
+  public ServerSideHttpLockManager(String requestUrl) {
+    initialize(ImmutableMap.of(REQUEST_URL, requestUrl));
+  }
+
+  @Override
+  public boolean acquire(String entityId, String ownerId) {
+    return process(entityId, ownerId, LOCK);
+  }
+
+  @Override
+  public void initialize(Map<String, String> properties) {
+    super.initialize(properties);
+    init(properties.getOrDefault(REQUEST_URL, null));
+  }
+
+  @Override
+  public void close() throws Exception {
+    super.close();
+    if (httpClient != null) {
+      httpClient.getConnectionManager().shutdown();
+    }
+  }
+
+  private synchronized void init(String requestUrl) {
+    if (requestUrl == null) {
+      String msg = String.format("[%s] must be set.", REQUEST_URL);
+      throw new IllegalArgumentException(msg);
+    }
+    if (this.httpUrl == null) {
+      this.httpUrl = requestUrl;
+    }
+    if (this.httpClient == null) {
+      DefaultHttpClient defaultHttpClient = new DefaultHttpClient();
+      defaultHttpClient.addRequestInterceptor(new RequestAcceptEncoding());
+      defaultHttpClient.addResponseInterceptor(new ResponseContentEncoding());
+      defaultHttpClient
+          .getParams()
+          .setParameter(CoreConnectionPNames.CONNECTION_TIMEOUT, acquireTimeoutMs());
+      defaultHttpClient
+          .getParams()
+          .setParameter(CoreConnectionPNames.SO_TIMEOUT, heartbeatTimeoutMs());
+      this.httpClient = new DefaultHttpClient();
+    }
+  }
+
+  private String encode(String entity) {
+    if (entity == null) {
+      return null;
+    }
+    return new String(
+        Base64.getUrlEncoder().encode(entity.getBytes(StandardCharsets.UTF_8)),
+        StandardCharsets.UTF_8);
+  }
+
+  private boolean process(String entityId, String ownerId, String operator) {
+    try {
+      HttpGet lockRequest = new HttpGet();
+      lockRequest.addHeader("Content-Type", "application/json");
+      HttpParams requestParams = new BasicHttpParams();
+      requestParams.setParameter(OWNER_ID, encode(ownerId));
+      requestParams.setParameter(ENTITY_ID, encode(entityId));
+      requestParams.setParameter(OPERATOR, operator);
+      String requestUrl =
+          new URIBuilder(httpUrl)
+              .setParameter(OWNER_ID, encode(ownerId))
+              .setParameter(ENTITY_ID, encode(entityId))
+              .setParameter(OPERATOR, operator)
+              .build()
+              .toString();
+      lockRequest.setParams(requestParams);
+      lockRequest.setURI(new URI(requestUrl));
+      HttpResponse response = httpClient.execute(lockRequest);
+      StatusLine stateLine = response.getStatusLine();
+      int statsCode = stateLine.getStatusCode();
+      lockRequest.abort();
+      return REQUEST_SUCCESS == statsCode;
```

Comment:
To be honest, I don't think we need to be concerned with the exact cause of the failure.Also, since we don't know what data the user intends to return, parsing the data returned by the user introduces a significant amount of logical complexity. Instead of doing this, it's easier to just detect the HTTP status code. It's not difficult for users to implement the rest-api service in the way we want.